samples: boards: Add support for nucleo_wba55cg board

lkoudelka opened the pull request:
run mco example also with nucleo_wba55cg board
due to gpio conflict uart print out not functional

Signed-off-by:Lubos Koudelka <lubos.koudelka@st.com>

FRASTM commented:
suggestion
```patch
diff --git a/samples/boards/st/mco/sample.yaml b/samples/boards/st/mco/sample.yaml
index bf8a97cad2a..ae76926f4ea 100644
--- a/samples/boards/st/mco/sample.yaml
+++ b/samples/boards/st/mco/sample.yaml
@@ -2,5 +2,5 @@ sample:
   name: STM32 microcontroller clock output (MCO) example
 tests:
   sample.board.stm32.mco:
-    platform_allow: nucleo_u5a5zj_q stm32f746g_disco
+    platform_allow: nucleo_u5a5zj_q stm32f746g_disco nucleo_wba55cg
     tags: board
```

FRASTM commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,34 @@
+ */
+#define MCO1_SEL_LSE 7
+
+/* See reference manual (RM0456):
```
! reference manual (RM0456) is not the one for the stm32wba55

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,34 @@
+	status = "disabled";
+};
+
+/* See reference manual (RM0493):
+ *   0b0111: LSE clock selected
+ */
+#define MCO1_SEL_LSE 7
```
Please remove these lines

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,34 @@
+
+&mco1 {
+	status = "okay";
+	clocks = <&rcc STM32_SRC_LSE MCO1_SEL(MCO1_SEL_LSE)>;
```
```suggestion
	clocks = <&rcc STM32_SRC_LSE MCO1_SEL(7)>;
```

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,34 @@
+/* See reference manual (RM0456):
+ *   0b001: MCO divided by 2
+ */
+#define MCO1_PRE_DIV_2 1
```
DIV definitions to be added in stm32wba_clock.h

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,34 @@
+/* The clock that is output must be enabled. */
```
Please add copyright.

lkoudelka committed:
dt-bindings: clock: Add MCO register definition for STM32WBA

Added MCO register definition to support MCO functionality in the STM32WBA
series. This update also includes the addition of MCO_PRE_DIV and MCO_SEL
defines.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

lkoudelka committed:
dts: arm: st: wba: Add MCO1 peripheral to STM32WBA device tree

Added MCO1 peripheral definition to the STM32WBA device tree to enable
MCO functionality.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

lkoudelka committed:
samples: boards: Add MCO support for nucleo_wba55cg

Added overlay and modified sample.yaml to support the nucleo_wba55cg
board in the MCO example. Due to GPIO conflict, UART print out is not
functional.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

erwango commented on include/zephyr/dt-bindings/clock/stm32wba_clock.h:
```diff
@@ -100,5 +100,29 @@
 #define ADC_SEL(val)		STM32_DOMAIN_CLOCK(val, 7, 12, CCIPR3_REG)
 /** BCDR1 devices */
 #define RTC_SEL(val)		STM32_DOMAIN_CLOCK(val, 3, 8, BCDR1_REG)
+/** @brief RCC_CFGRx register offset */
```
Not blocking: Please break a line before this.

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,30 @@
+&clk_lse {
+	status = "okay";
+};
+/* MCO use same pin like usart1 RX - disable usart */
```
Not blocking: Please break a line before this.

erwango reviewed (approved):
2 non blocking comments, otherwise LGTM

kartben commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,30 @@
+&mco1 {
+	status = "okay";
+	clocks = <&rcc STM32_SRC_LSE MCO1_SEL(MCO_SEL_LSE)>;
+	prescaler = <MCO1_PRE(MCO1_PRE_DIV_2)>;
```
MCO1_PRE_DIV_2 --> MCO_PRE_DIV_2 

see https://github.com/zephyrproject-rtos/zephyr/pull/85860
